### DISCUSSION TITLE
Best way to change embedding?

### DISCUSSION THREAD

**Discussion by hiepxanh** [qdrant/qdrant#3797] [Q&A] [answered]
there is two case I'm working with Qdrant:
1) I'm changing from embedding model A to model B which have bigger dimension, reason is it's cheaper, or better quality.
2) I'm testing hybrid search using sparse vectors

So the plan is:  scan all points, then batch by batch get point value, then push it back:
-  step 0: but since collection API is fixed, so I need to create new collection with new vector support. https://qdrant.tech/documentation/concepts/collections/#update-collection-parameters
- step 1: recreate new collection with 2 vectors: https://qdrant.tech/documentation/concepts/collections/#collection-with-multiple-vectors
- step 2: prepare embedding API for sprare vector or new dense vector to embedding
- step 3: batch data to server by: add [new point](https://qdrant.tech/documentation/concepts/points/#upload-points) or [update vector](https://qdrant.tech/documentation/concepts/points/#update-vectors)? 

is that a good strategy? do I have any better solution to generate new vector? I would like to have a new article from qdrant team to guide us to generate spare vector or changing new embedding vector from current data

**Comment by agourlay** [comment, marked as answer]
@hiepxanh Hi :wave: 

I think your strategy is the right way.

Doing incremental scrolling with transfer into a new collection with batch upload points.

You can play with the scroll/batch size for better performance.

**Reply by hiepxanh** [reply]
thank you, so much <3
